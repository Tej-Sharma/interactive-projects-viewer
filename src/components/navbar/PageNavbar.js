import React, { Component } from "react";
import Navbar from "react-bootstrap/Navbar";
import Nav from "react-bootstrap/Nav";
import logo from "../../example_studio42_logo.svg";
import Zoom from 'react-reveal/Zoom';

export class PageNavbar extends Component {
  render() {
    return (
      <div>
        <Navbar collapseOnSelect bg="dark" variant="dark" fixed="top" >
          <Navbar.Brand href="#home">
            <img
              alt=""
              src={logo}
              width="45"
              height="45"
              className="d-inline-block align-top"
            />{" "}
            Projects
          </Navbar.Brand>
          <Navbar.Toggle aria-controls="responsive-navbar-nav" />
          <Navbar.Collapse id="responsive-navbar-nav">
            <Nav className="mr-auto"></Nav>
            <Nav>
              <Zoom right>
                <Nav.Link
                  target="_blank"
                  href="https://github.com/Trollgen-Studios"
                >
                  Made by Tejas
                </Nav.Link>
              </Zoom>
            </Nav>
          </Navbar.Collapse>
        </Navbar>
      </div>
    );
  }
}

export default PageNavbar;
