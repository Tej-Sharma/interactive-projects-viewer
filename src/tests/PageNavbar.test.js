import React from "react";
import { shallow } from "enzyme";
import PageNavbar from "../components/navbar/PageNavbar";
import renderer from "react-test-renderer";

// Not much to test here, nevertheless, ensure that the correct text exists
// And do snapshot testing to ensure that no future problems suddenly come up
describe("PageNavbar tests", () => {
  // Snapshot test
  test("PageNavbar renders correctly", () => {
    const tree = renderer.create(<PageNavbar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // Unit test
  test("Correct text exists", () => {
    const wrapper = shallow(<PageNavbar />);
    expect(wrapper.text()).toContain("Our ProjectsMade by Tejas");
  });
});
