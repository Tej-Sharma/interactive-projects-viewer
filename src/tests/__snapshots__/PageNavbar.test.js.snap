// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageNavbar tests PageNavbar renders correctly 1`] = `
<div>
  <nav
    className="navbar navbar-expand navbar-dark bg-dark fixed-top"
  >
    <a
      className="navbar-brand"
      href="#home"
    >
      <img
        alt=""
        className="d-inline-block align-top"
        height="45"
        src="example_studio42_logo.svg"
        width="45"
      />
       
      Projects
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="mr-auto navbar-nav"
        onKeyDown={[Function]}
      />
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <a
            className="nav-link"
            data-rb-event-key="https://github.com/Trollgen-Studios"
            href="https://github.com/Trollgen-Studios"
            onClick={[Function]}
            onKeyDown={[Function]}
            target="_blank"
          >
            Made by Tejas
          </a>
        </div>
      </div>
    </div>
  </nav>
</div>
`;
